drivers/syslog/ramlog.c

```diff
@@ -287,28 +274,64 @@ static void ramlog_addchar(FAR struct ramlog_dev_s *priv, char ch)
 static ssize_t ramlog_addbuf(FAR struct ramlog_dev_s *priv,
                              FAR const char *buffer, size_t len)
 {
+#if defined(CONFIG_RAMLOG_SYSLOG) || defined(CONFIG_RAMLOG_CRLF)
+  FAR struct ramlog_header_s *header = priv->rl_header;
+#endif
   irqstate_t flags;
-  size_t nwritten;
-  char ch;
 
   /* Disable interrupts (in case we are NOT called from interrupt handler) */
 
   flags = enter_critical_section();
 
-  for (nwritten = 0; nwritten < len; nwritten++)
+#ifdef CONFIG_RAMLOG_SYSLOG
+  if (header->rl_magic != RAMLOG_MAGIC_NUMBER && priv == &g_sysdev)
+    {
+      memset(header, 0, sizeof(g_sysbuffer));
+      header->rl_magic = RAMLOG_MAGIC_NUMBER;
+    }
+#endif
+
+  if (len > priv->rl_bufsize)
     {
-      /* Get the next character to output */
+      buffer += (len - priv->rl_bufsize);
+      len = priv->rl_bufsize;
+    }
+
+#ifdef CONFIG_RAMLOG_CRLF
+  FAR char *rlbuf = header->rl_buffer;
```

Comment:
```suggestion
  FAR char *buf = header->rl_buffer;
```
move the declaration to the begin of block.

Reply:
Done